src/crypto/hsm/infineon/CHIPCryptoPALHsm_HKDF_trustm.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+*
+* Copyright (c) 2021 Project CHIP Authors
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/**
+* @file
+* HSM based implementation of CHIP crypto primitives
+* Based on configurations in CHIPCryptoPALHsm_config.h file,
+* chip crypto apis use either HSM or rollback to software implementation.
+*/
+
+#include "CHIPCryptoPALHsm_utils_trustm.h"
+#include <lib/core/CHIPEncoding.h>
+#include "optiga_crypt.h"
+#include "optiga/optiga_util.h"
+#include "optiga_lib_types.h"
+
+
+static const uint8_t metadata [] = {
+//Metadata tag in the data object
+0x20, 0x06,
+//Data object type set to PRESSEC
+0xE8, 0x01, 0x21,
+0xD3, 0x01, 0x00,
+};
+
+#if ENABLE_HSM_HKDF_SHA256
+
+namespace chip {
+namespace Crypto {
+
+HKDF_shaHSM::HKDF_shaHSM(){}
+HKDF_shaHSM::~HKDF_shaHSM() {}

```

Comment:
Please do not add defaults for constructors/destructors in .cpp file. Can use `=default` in header.

Reply:
Hi @tcarmelveilleux, Thanks for your review. We have fixed this in the latest commit.